### PR TITLE
[Datastore] Allow passing GCS credentials as a json string

### PR DESCRIPTION
**Pull request by alxtkr77**
https://jira.iguazeng.com/browse/ML-5451

### CODE REVIEW

File: mlrun/datastore/google_cloud_storage.py [hunk 1]
```diff
@@ -84,7 +51,30 @@ def get_filesystem(self, silent=True):
         return self._filesystem
 
     def get_storage_options(self):
-        return dict(token=self._get_secret_or_env("GOOGLE_APPLICATION_CREDENTIALS"))
+        google_credentials = self._get_secret_or_env("GOOGLE_APPLICATION_CREDENTIALS")
+        gcp_credentials = self._get_secret_or_env("GCP_CREDENTIALS")
+        if (
+            gcp_credentials
+            and google_credentials
+            and google_credentials != gcp_credentials
```
Comment (tomerm-iguazio) [inline mlrun/datastore/google_cloud_storage.py:59]: In `google_credentials` you Will get a path to a credential file.
They should never be the same!

If you want, you can read the file and compare its data.

File: tests/system/datastore/test_google_cloud_storage.py [hunk 2]
```diff
@@ -60,7 +62,18 @@ def setup_class(cls):
         cls.credentials_path = test_environment["GOOGLE_APPLICATION_CREDENTIALS"]
         cls.test_dir = "test_mlrun_gcs_system_objects"
         cls.profile_name = "gcs_system_profile"
-        cls._gcs_fs = fsspec.filesystem("gcs", token=cls.credentials_path)
+        try:
```
Comment (tomerm-iguazio) [inline tests/system/datastore/test_google_cloud_storage.py:65]: why try and except is needed?

File: mlrun/datastore/google_cloud_storage.py [hunk 3]
```diff
@@ -84,7 +51,30 @@ def get_filesystem(self, silent=True):
...
+        ):
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Both GOOGLE_APPLICATION_CREDENTIALS and GCP_CREDENTIALS are set but not the same"
+            )
+        credentials = google_credentials or gcp_credentials
+        if credentials:
+            try:
+                credentials = json.loads(credentials)
```
Comment (tomerm-iguazio) [inline mlrun/datastore/google_cloud_storage.py:67]: why do you need to use try and except?

Comment (tomerm-iguazio) [reply mlrun/datastore/google_cloud_storage.py:67]: json.loads will not work on a path to json file, as GOOGLE_APPLICATION_CREDENTIALS.

**Comment by tomerm-iguazio** [comment]
image
image

Every test failed in mlrun/tests/integration/google_cloud_storage/test_google_cloud_storage.py.
3 tests failed in mlrun/tests/system/datastore/test_google_cloud_storage.py.
**Please open a PR only after checking all relevant tests** (these integration and system tests are not running in the ci):

`mlrun/tests/system/datastore/test_google_cloud_storage.py`
`mlrun/tests/integration/google_cloud_storage/test_google_cloud_storage.py`

**Review by tomerm-iguazio — CHANGES_REQUESTED** [review]

**Review by assaf758 — APPROVED** [review]

**Pull request merged**